frontend/test/metabase/scenarios/question/filter.cy.spec.js

```diff
@@ -1030,4 +1030,50 @@ describe("scenarios > question > filter", () => {
         .and("contain", "Average of Total");
     }
   });
+
+  describe.skip("specific combination of filters can cause frontend reload or blank screen (metabase#16198)", () => {
+    it("shouldn't display chosen category in a breadcrumb (metabase#16198-1)", () => {
+      visitQuestionAdhoc({
+        dataset_query: {
+          database: 1,
+          query: {
+            "source-table": PRODUCTS_ID,
+            filter: [
+              "and",
+              ["=", ["field", PRODUCTS.CATEGORY, null], "Gizmo"],
+              ["=", ["field", PRODUCTS.ID, null], 1],
+            ],
+          },
+          type: "query",
+        },
+      });
+
+      cy.findByRole("link", { name: "Sample Dataset" })
+        .parent()
+        .within(() => {
+          cy.findByText("Gizmo").should("not.exist");
+        });
+    });
+
+    it("adding an ID filter shouldn't cause page error and page reload (metabase#16198-2)", () => {
+      openOrdersTable({ mode: "notebook" });
+      cy.findByText("Filter").click();
+      cy.findByText("Custom Expression").click();
+      cy.get("[contenteditable=true]")
+        .type("[Total] < [Product → Price]")
+        .blur();
+      cy.button("Done").click();
+      cy.findByText(/^Total is less than/);
```

Comment:
and line 1075

This might be going to cause a little friction with master/future. Might become `Total < Price` #16174
It would probably be better just to detect if there's a filter element instead of the text.

Also, just curious, is `cy.findByText(/^Total is less than/);` automatically the same as `cy.findByText(/^Total is less than/).should("exist");` ?

Reply:
> This might be going to cause a little friction with master/future. Might become `Total < Price` #16174
> It would probably be better just to detect if there's a filter element instead of the text.

I'll try to see if there's some non-ambiguous way to do that.

> Also, just curious, is `cy.findByText(/^Total is less than/);` automatically the same as `cy.findByText(/^Total is less than/).should("exist");` ?

Yup. Implicit vs explicit assertion.

Reply:
@flamber 
Unfortunately, there isn't a single reliable thing that I can grab onto and simply assert that the filter exist. We'll have to rely on the string.
The best I can do now is use `cy.findByText(/^Total/);`
![image](https://user-images.githubusercontent.com/31325167/119484696-8ddf7080-bd56-11eb-8330-802d4464f025.png)

Please LMK if you're more for that version and I'll update accordingly.